tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -628,6 +628,38 @@ public function testOneBelongsToManyJoinDataAssociated(): void
             'joinData should contain a user entity.',
         );
         $this->assertSame('Mark', $result->tags[1]->_joinData->user->username);
+
+        $data = [
+            'title' => 'My title',
+            'body' => 'My content',
+            'author_id' => 1,
+            'tags' => [
+                [
+                    'tag' => 'news',
+                    'junction' => [
+                        'active' => 1,
+                        'user' => ['username' => 'Bill'],
+                    ],
+                ],
+                [
+                    'tag' => 'cakephp',
+                    'junction' => [
+                        'active' => 0,
+                        'user' => ['username' => 'Mark'],
+                    ],
+                ],
+            ],
+        ];
+
+        $this->articles->Tags->setJunctionProperty('junction');
+
+        $marshall = new Marshaller($this->articles);
+        $result = $marshall->one($data, ['associated' => ['Tags.junction.Users']]);
```

Comment:
We should have a test case for patching existing records with a custom junction property as well.

Reply:
Additional tests added in `testMergeBelongsToManyJoinData()` using `Marshaller::merge()`.